### PR TITLE
pkg/obfuscate: fix SQL tokenizer number parsing

### PR DESCRIPTION
**Pull request by knusbaum**
scanNumber failed to parse certain negative numbers and also failed to
set an error in the tokenizer when it returned LexError, resulting in
`nil` error messages. This commit causes scanNumber to correctly parse
those numbers and also set a more meaningful error message when it
encounters an error.

This also adds some tests specifically for the number parser.

### CODE REVIEW

**Comment by knusbaum** [comment]
I'll add a regression testcase for this.

**Review by gbbr — APPROVED** [review]

File: pkg/obfuscate/sql_tokenizer.go [hunk 1]
```diff
@@ -634,7 +634,7 @@ func (tkn *SQLTokenizer) scanNumber(seenDecimalPoint bool) (TokenKind, []byte) {
 			}
 			// octal int
 			if seenDecimalDigit {
-				// tkn.setErr called in caller
+				tkn.setErr("invalid number")
```
Comment (gbbr) [inline pkg/obfuscate/sql_tokenizer.go:637]: Can we make a more specific error message? Why is it an invalid number? If we end up debugging this, we'll have an easier time knowing where the error occurred vs. having two places with the same error message.

Comment (knusbaum) [reply pkg/obfuscate/sql_tokenizer.go]: Done.

File: pkg/obfuscate/sql_test.go [hunk 2]
```diff
@@ -1943,6 +1942,41 @@ func TestUnicodeDigit(t *testing.T) {
 	o.ObfuscateSQLString(hangStr)
 }
 
+func TestParseNumber(t *testing.T) {
+	var testCases = []struct {
+		in       string
+		expected string
+		error    string
```
Comment (ajgajg1134) [inline pkg/obfuscate/sql_test.go:1949]: Error appears unused so it can be removed

**Review by ajgajg1134 — APPROVED** [review]
Looks good!

**Review by buraizu — APPROVED** [review]

**Pull request merged**